src/test/java/cloud/fogbow/ras/core/cloudconnector/LocalCloudConnectorTest.java

```diff
@@ -1,942 +1,1254 @@
+
+        String expected = Messages.Exception.NULL_VALUE_RETURNED;
+
+        try {
+            // exercise
+            this.localCloudConnector.requestInstance(order);
+            Assert.fail();
+        } catch (UnexpectedException e) {
+            // verify
+            Assert.assertEquals(expected, e.getMessage());
+            Mockito.verify(this.localCloudConnector, Mockito.times(BaseUnitTests.RUN_ONCE)).auditRequest(
+                    Mockito.eq(Operation.CREATE), Mockito.eq(ResourceType.PUBLIC_IP), Mockito.any(SystemUser.class),
+                    Mockito.anyString());
+        }
     }
+    
+    // test case: When invoking the getInstance method with a valid compute order it
+    // must call the doGetInstance method, and confirm in auditRequest the GET
+    // operation and the COMPUTE resource type.
+    @Test
+    public void testGetInstanceWithComputeOrder() throws FogbowException {
+        // set up
+        Order order = createLocalComputeOrder();
+        order.setInstanceId(FAKE_INSTANCE_ID);
+        order.setOrderState(OrderState.FULFILLED);
+
+        CloudUser cloudUser = Mockito.mock(CloudUser.class);
+        Mockito.when(this.mapperPlugin.map(Mockito.any(SystemUser.class))).thenReturn(cloudUser);
 
-    // test case: If plugin returns a null instance id, the method requestInstance() must throw an exception
-    @Test(expected = UnexpectedException.class)
-    public void testExceptionNullAttachmentInstanceId() throws FogbowException {
-        // set up
-        ComputeOrder source = Mockito.mock(ComputeOrder.class);
-        VolumeOrder target = Mockito.mock(VolumeOrder.class);
-        Mockito.when(source.getSystemUser()).thenReturn(this.systemUser);
-        Mockito.when(source.getProvider()).thenReturn(FAKE_PROVIDER);
-        Mockito.when(target.getSystemUser()).thenReturn(this.systemUser);
-        Mockito.when(target.getProvider()).thenReturn(FAKE_PROVIDER);
-        SharedOrderHolders.getInstance().getActiveOrdersMap().put(FAKE_COMPUTE_ID, source);
-        SharedOrderHolders.getInstance().getActiveOrdersMap().put(FAKE_VOLUME_ID, target);
-        this.order = Mockito.mock(AttachmentOrder.class);
-        Mockito.when(this.order.getSystemUser()).thenReturn(this.systemUser);
-        Mockito.when(this.order.getProvider()).thenReturn(FAKE_PROVIDER);
-        Mockito.when(((AttachmentOrder) this.order).getComputeOrderId()).thenReturn(FAKE_COMPUTE_ID);
-        Mockito.when(((AttachmentOrder) this.order).getVolumeOrderId()).thenReturn(FAKE_VOLUME_ID);
-        Mockito.when(this.order.getType()).thenReturn(ResourceType.ATTACHMENT);
-        Mockito.when(attachmentPlugin.requestInstance(Mockito.any(AttachmentOrder.class), Mockito.any(CloudUser.class))).thenReturn(null);
+        OrderInstance instance = new ComputeInstance(FAKE_INSTANCE_ID);
+        Mockito.doReturn(instance).when(this.localCloudConnector).getResourceInstance(Mockito.eq(order),
```

Comment:
you shouldn't mock it here. This method is called inside doGetInstance. Its enough to test if doGetInstance, instance.setState and auditRequest have been called in this test for the successful case.

Reply:
same thing to the other tests

Reply:
Okay, but anyway you should mock the doGetInstance method instead of getResourceInstance otherwise a NullPontException will be thrown.

I'll fix these tests and verify the need to add others for testing the getResourceInstance method separately.